src/cdomains/apron/sharedFunctions.apron.ml

```diff
@@ -285,11 +287,38 @@ struct
     !expr
 
 
+  let lcm_den linexpr1 =
+    let exception UnsupportedScalar 
+    in
+    let frac_of_scalar scalar =
+      if Scalar.is_infty scalar <> 0 then (* infinity means unbounded *)
+        None
+      else match scalar with
+        | Float f -> if Stdlib.Float.is_integer f then Some (Q.of_float f) else None
+        | Mpqf f -> Some (Z_mlgmpidl.q_of_mpqf f)
+        | _ -> raise UnsupportedScalar
+    in
+    let extract_den (c:Coeff.union_5) _ =
+      match c with
+      | Scalar c -> BatOption.map (fun q -> Q.den q) (frac_of_scalar c)
+      | _ -> None
+    in
+    let lcm_denom = ref (BatOption.default Z.one (extract_den (Linexpr1.get_cst linexpr1) ())) in
+    let lcm_coeff (c:Coeff.union_5) v =
+      match (extract_den c v) with
+      | Some z -> lcm_denom := Z.lcm z !lcm_denom
+      | _      -> ()
+    in
```

Comment:
Do we really need the `ref` here?

Reply:
I guess we do, the API doesn't really provide anything helpful here:

https://antoinemine.github.io/Apron/doc/api/ocaml/Linexpr1.html

Reply:
I did not like the ref either, but did not immediately see a way to easily avoid it. At least it is contained in a small area